zebra-state/src/service.rs

```diff
@@ -252,6 +252,87 @@ impl StateService {
     pub fn utxo(&self, outpoint: &transparent::OutPoint) -> Option<transparent::Output> {
         self.mem.utxo(outpoint).or_else(|| self.sled.utxo(outpoint))
     }
+
+    /// Return an iterator over the relevant chain of the block identified by
+    /// `hash_or_height`.
+    #[allow(dead_code)]
+    pub fn chain(&self, hash_or_height: HashOrHeight) -> Iter<'_> {
```

Comment:
Suggestion:
These functions use any chain, but the rest of the functions in the service use the best chain.
Should we rename them to make that clearer?

```suggestion
    pub fn utxo_any_chain(&self, outpoint: &transparent::OutPoint) -> Option<transparent::Output> {
        self.mem.utxo(outpoint).or_else(|| self.sled.utxo(outpoint))
    }

    /// Return an iterator over the relevant chain of the block identified by
    /// `hash_or_height`.
    #[allow(dead_code)]
    pub fn any_chain(&self, hash_or_height: HashOrHeight) -> Iter<'_> {
```

If we do this rename, we should also rename the memory and state `utxo` methods.

Reply:
Hmm, actually, hashes use any chain, and height uses the best chain.

Let's split the hash and height into different methods?
(I don't need the height method right now, but it seems useful in future.)

Reply:
I feel like this discussion should be split into a different PR / issue. I had a similar discussion with @hdevalence when I was pairing on the block locator fix where we did some cleanup to these methods to make it uniform across the finalized state, non-finalized state, and service. I don't remember precisely what points were brought up though so we should probably split this into a complete and separate discussion about how to structure our internal APIs.

Reply:
If that's what you decided, that's fine with me. I don't really need to repeat that discussion.